File: snippets/javascript/src/querySuggestions.ts
/* eslint no-console: ["error", { allow: ["log"] }] */

import { querySuggestionsClient } from '@algolia/client-query-suggestions';

// Snippet for the createConfig method.
//
// createConfig0
export async function snippetForcreateConfig(): Promise<void> {
  // >SEPARATOR createConfig
  // Initialize the client
  const client = querySuggestionsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.createConfig({
    indexName: 'theIndexName',
    sourceIndices: [
      {
        indexName: 'testIndex',
        facets: [{ attribute: 'test' }],
        generate: [['facetA', 'facetB'], ['facetC']],
      },
    ],
    languages: ['french'],
    exclude: ['test'],
  });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the customDelete method.
//
// allow del method for a custom path with minimal parameters
export async function snippetForcustomDelete(): Promise<void> {
  // >SEPARATOR customDelete
  // Initialize the client
  const client = querySuggestionsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.customDelete({ path: '/test/minimal' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the customGet method.
//
// allow get method for a custom path with minimal parameters
export async function snippetForcustomGet(): Promise<void> {
  // >SEPARATOR customGet
  // Initialize the client
  const client = querySuggestionsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.customGet({ path: '/test/minimal' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the customPost method.
//
// allow post method for a custom path with minimal parameters
export async function snippetForcustomPost(): Promise<void> {
  // >SEPARATOR customPost
  // Initialize the client
  const client = querySuggestionsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.customPost({ path: '/test/minimal' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the customPut method.
//
// allow put method for a custom path with minimal parameters
export async function snippetForcustomPut(): Promise<void> {
  // >SEPARATOR customPut
  // Initialize the client
  const client = querySuggestionsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.customPut({ path: '/test/minimal' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the deleteConfig method.
//
// deleteConfig0
export async function snippetFordeleteConfig(): Promise<void> {
  // >SEPARATOR deleteConfig
  // Initialize the client
  const client = querySuggestionsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.deleteConfig({ indexName: 'theIndexName' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getAllConfigs method.
//
// getAllConfigs0
export async function snippetForgetAllConfigs(): Promise<void> {
  // >SEPARATOR getAllConfigs
  // Initialize the client
  const client = querySuggestionsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getAllConfigs();

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getConfig method.
//
// Retrieve QS config e2e
export async function snippetForgetConfig(): Promise<void> {
  // >SEPARATOR getConfig
  // Initialize the client
  const client = querySuggestionsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getConfig({
    indexName: 'cts_e2e_browse_query_suggestions',
  });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getConfigStatus method.
//
// getConfigStatus0
export async function snippetForgetConfigStatus(): Promise<void> {
  // >SEPARATOR getConfigStatus
  // Initialize the client
  const client = querySuggestionsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getConfigStatus({ indexName: 'theIndexName' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the getLogFile method.
//
// getLogFile0
export async function snippetForgetLogFile(): Promise<void> {
  // >SEPARATOR getLogFile
  // Initialize the client
  const client = querySuggestionsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.getLogFile({ indexName: 'theIndexName' });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

// Snippet for the updateConfig method.
//
// updateConfig0
export async function snippetForupdateConfig(): Promise<void> {
  // >SEPARATOR updateConfig
  // Initialize the client
  const client = querySuggestionsClient(
    'YOUR_APP_ID',
    'YOUR_API_KEY',
    'YOUR_APP_ID_REGION'
  );

  // Call the API
  const response = await client.updateConfig({
    indexName: 'theIndexName',
    querySuggestionsConfiguration: {
      sourceIndices: [
        {
          indexName: 'testIndex',
          facets: [{ attribute: 'test' }],
          generate: [['facetA', 'facetB'], ['facetC']],
        },
      ],
      languages: ['french'],
      exclude: ['test'],
    },
  });

  // use typed response
  console.log(response);
  // SEPARATOR<
}

File: snippets/csharp/src/QuerySuggestions.cs
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.QuerySuggestions;
using Action = Algolia.Search.Models.Search.Action;

public class SnippetQuerySuggestionsClient
{
  /// <summary>
  /// Snippet for the CreateConfig method.
  ///
  /// createConfig0
  /// </summary>
  public async Task SnippetForQuerySuggestionsClientCreateConfig()
  {
    // >SEPARATOR createConfig
    // Initialize the client
    var client = new QuerySuggestionsClient(
      new QuerySuggestionsConfig("YOUR_APP_ID", "YOUR_API_KEY", "YOUR_APP_ID_REGION")
    );

    // Call the API
    var response = await client.CreateConfigAsync(
      new QuerySuggestionsConfigurationWithIndex
      {
        IndexName = "theIndexName",
        SourceIndices = new List<SourceIndex>
        {
          new SourceIndex
          {
            IndexName = "testIndex",
            Facets = new List<Facet> { new Facet { Attribute = "test", } },
            Generate = new List<List<string>>
            {
              new List<string> { "facetA", "facetB" },
              new List<string> { "facetC" }
            },
          }
        },
        Languages = new Languages(new List<string> { "french" }),
        Exclude = new List<string> { "test" },
      }
    );
    // SEPARATOR<
  }

  /// <summary>
  /// Snippet for the CustomDelete method.
  ///
  /// allow del method for a custom path with minimal parameters
  /// </summary>
  public async Task SnippetForQuerySuggestionsClientCustomDelete()
  {
    // >SEPARATOR customDelete
    // Initialize the client
    var client = new QuerySuggestionsClient(
      new QuerySuggestionsConfig("YOUR_APP_ID", "YOUR_API_KEY", "YOUR_APP_ID_REGION")
    );

    // Call the API
    var response = await client.CustomDeleteAsync("/test/minimal");
    // SEPARATOR<
  }

  /// <summary>
  /// Snippet for the CustomGet method.
  ///
  /// allow get method for a custom path with minimal parameters
  /// </summary>
  public async Task SnippetForQuerySuggestionsClientCustomGet()
  {
    // >SEPARATOR customGet
    // Initialize the client
    var client = new QuerySuggestionsClient(
      new QuerySuggestionsConfig("YOUR_APP_ID", "YOUR_API_KEY", "YOUR_APP_ID_REGION")
    );

    // Call the API
    var response = await client.CustomGetAsync("/test/minimal");
    // SEPARATOR<
  }

  /// <summary>
  /// Snippet for the CustomPost method.
  ///
  /// allow post method for a custom path with minimal parameters
  /// </summary>
  public async Task SnippetForQuerySuggestionsClientCustomPost()
  {
    // >SEPARATOR customPost
    // Initialize the client
    var client = new QuerySuggestionsClient(
      new QuerySuggestionsConfig("YOUR_APP_ID", "YOUR_API_KEY", "YOUR_APP_ID_REGION")
    );

    // Call the API
    var response = await client.CustomPostAsync("/test/minimal");
    // SEPARATOR<
  }

  /// <summary>
  /// Snippet for the CustomPut method.
  ///
  /// allow put method for a custom path with minimal parameters
  /// </summary>
  public async Task SnippetForQuerySuggestionsClientCustomPut()
  {
    // >SEPARATOR customPut
    // Initialize the client
    var client = new QuerySuggestionsClient(
      new QuerySuggestionsConfig("YOUR_APP_ID", "YOUR_API_KEY", "YOUR_APP_ID_REGION")
    );

    // Call the API
    var response = await client.CustomPutAsync("/test/minimal");
    // SEPARATOR<
  }

  /// <summary>
  /// Snippet for the DeleteConfig method.
  ///
  /// deleteConfig0
  /// </summary>
  public async Task SnippetForQuerySuggestionsClientDeleteConfig()
  {
    // >SEPARATOR deleteConfig
    // Initialize the client
    var client = new QuerySuggestionsClient(
      new QuerySuggestionsConfig("YOUR_APP_ID", "YOUR_API_KEY", "YOUR_APP_ID_REGION")
    );

    // Call the API
    var response = await client.DeleteConfigAsync("theIndexName");
    // SEPARATOR<
  }

  /// <summary>
  /// Snippet for the GetAllConfigs method.
  ///
  /// getAllConfigs0
  /// </summary>
  public async Task SnippetForQuerySuggestionsClientGetAllConfigs()
  {
    // >SEPARATOR getAllConfigs
    // Initialize the client
    var client = new QuerySuggestionsClient(
      new QuerySuggestionsConfig("YOUR_APP_ID", "YOUR_API_KEY", "YOUR_APP_ID_REGION")
    );

    // Call the API
    var response = await client.GetAllConfigsAsync();
    // SEPARATOR<
  }

  /// <summary>
  /// Snippet for the GetConfig method.
  ///
  /// Retrieve QS config e2e
  /// </summary>
  public async Task SnippetForQuerySuggestionsClientGetConfig()
  {
    // >SEPARATOR getConfig
    // Initialize the client
    var client = new QuerySuggestionsClient(
      new QuerySuggestionsConfig("YOUR_APP_ID", "YOUR_API_KEY", "YOUR_APP_ID_REGION")
    );

    // Call the API
    var response = await client.GetConfigAsync("cts_e2e_browse_query_suggestions");
    // SEPARATOR<
  }

  /// <summary>
  /// Snippet for the GetConfigStatus method.
  ///
  /// getConfigStatus0
  /// </summary>
  public async Task SnippetForQuerySuggestionsClientGetConfigStatus()
  {
    // >SEPARATOR getConfigStatus
    // Initialize the client
    var client = new QuerySuggestionsClient(
      new QuerySuggestionsConfig("YOUR_APP_ID", "YOUR_API_KEY", "YOUR_APP_ID_REGION")
    );

    // Call the API
    var response = await client.GetConfigStatusAsync("theIndexName");
    // SEPARATOR<
  }

  /// <summary>
  /// Snippet for the GetLogFile method.
  ///
  /// getLogFile0
  /// </summary>
  public async Task SnippetForQuerySuggestionsClientGetLogFile()
  {
    // >SEPARATOR getLogFile
    // Initialize the client
    var client = new QuerySuggestionsClient(
      new QuerySuggestionsConfig("YOUR_APP_ID", "YOUR_API_KEY", "YOUR_APP_ID_REGION")
    );

    // Call the API
    var response = await client.GetLogFileAsync("theIndexName");
    // SEPARATOR<
  }

  /// <summary>
  /// Snippet for the UpdateConfig method.
  ///
  /// updateConfig0
  /// </summary>
  public async Task SnippetForQuerySuggestionsClientUpdateConfig()
  {
    // >SEPARATOR updateConfig
    // Initialize the client
    var client = new QuerySuggestionsClient(
      new QuerySuggestionsConfig("YOUR_APP_ID", "YOUR_API_KEY", "YOUR_APP_ID_REGION")
    );

    // Call the API
    var response = await client.UpdateConfigAsync(
      "theIndexName",
      new QuerySuggestionsConfiguration
      {
        SourceIndices = new List<SourceIndex>
        {
          new SourceIndex
          {
            IndexName = "testIndex",
            Facets = new List<Facet> { new Facet { Attribute = "test", } },
            Generate = new List<List<string>>
            {
              new List<string> { "facetA", "facetB" },
              new List<string> { "facetC" }
            },
          }
        },
        Languages = new Languages(new List<string> { "french" }),
        Exclude = new List<string> { "test" },
      }
    );
    // SEPARATOR<
  }
}
